behaviour/graql/language/undefine.feature

```diff
@@ -260,45 +1218,286 @@ Feature: Graql Undefine Query
       | x      |
       | RULE   |
       | A_RULE |
-    Then graql undefine
+    When graql undefine
       """
       undefine a-rule sub rule;
       """
+    Then the integrity is validated
     Then get answers of graql query
       """
-      match $x sub rule; get;  
+      match $x sub rule; get;
       """
     Then answer size is: 1
 
 
-  Scenario: undefine a supertype errors if subtypes exist
+  Scenario: after undefining a rule, concepts previously inferred by that rule are no longer inferred
```

Comment:
what happens if you remove a rule in a tx without committing and then query something that it would infer?

Reply:
Will add test

Reply:
As it turns out, concepts inferred by the rule ARE retrieved if you don't commit the undefine!

Reply:
Added `Scenario: after undefining a rule without a commit, concepts previously inferred by that rule are still inferred`